handler/user_handler.go

```diff
@@ -93,3 +94,132 @@ func ListUsersHandler(userService service.UserService) http.HandlerFunc {
 		w.Write(responsebytes)
 	}
 }
+
+func UpdateUsersHandler(userService service.UserService) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		w.Header().Set("Content-Type", "application/json")
+
+		params := mux.Vars(r)
+		id, errInConversion := strconv.Atoi(params["id"])
+
+		if errInConversion != nil {
+			fmt.Printf("handler: Error while parameter conversion. Error: %s", errInConversion)
+			w.WriteHeader(http.StatusBadRequest)
+			responseBytes, err := json.Marshal(contract.ErrorResponse{Error: "Error while parameter conversion"})
+			w.Write(responseBytes)
+
+			if err != nil {
+				fmt.Printf("handler: Error while Marshal,%s", err)
+			}
+			return
+
+		}
+
+		var req contract.UpdateUserRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+
+		if err != nil {
+			fmt.Printf("handler:Error while decoding request for update, %s", err)
+			w.WriteHeader(http.StatusBadRequest)
+			responseBytes, err := json.Marshal(contract.ErrorResponse{Error: "invalid request"})
+			w.Write(responseBytes)
+
+			if err != nil {
+				fmt.Printf("handler: Error while Marshal,%s", err)
+			}
+			return
+		}
+
+		user, err := userService.UpdateUser(r.Context(), id, req)
+
+		if err == customerrors.UserDoesNotExist {
```

Comment:
err check should be in this order
if err != nil {
if err == customererr {
log error
return
}
log error 
return
}

pls make this change wherever neccessary